contracts/cw20-staked-balance-voting/src/contract.rs

```diff
@@ -0,0 +1,257 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Reply, Response, StdResult, SubMsg,
+    Uint128, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw_utils::parse_reply_instantiate_data;
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg, StakingInfo, TokenInfo};
+use crate::state::{
+    STAKING_CONTRACT, STAKING_CONTRACT_CODE_ID, STAKING_CONTRACT_UNSTAKING_DURATION, TOKEN,
+};
+
+const CONTRACT_NAME: &str = "crates.io:cw20-staked-balance-voting";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const INSTANTIATE_TOKEN_REPLY_ID: u64 = 0;
+const INSTANTIATE_STAKING_REPLY_ID: u64 = 1;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    match msg.token_info {
+        TokenInfo::Existing {
+            address,
+            staking_contract,
+        } => {
+            let address = deps.api.addr_validate(&address)?;
+            TOKEN.save(deps.storage, &address)?;
+
+            match staking_contract {
+                StakingInfo::Existing {
+                    staking_contract_address,
+                } => {
+                    let staking_contract_address =
+                        deps.api.addr_validate(&staking_contract_address)?;
+                    let resp: stake_cw20::msg::GetConfigResponse = deps.querier.query_wasm_smart(
+                        &staking_contract_address,
+                        &stake_cw20::msg::QueryMsg::GetConfig {},
+                    )?;
+
+                    if address != resp.token_address {
+                        return Err(ContractError::StakingContractMismatch {});
+                    }
+
+                    STAKING_CONTRACT.save(deps.storage, &staking_contract_address)?;
+                    Ok(Response::default()
+                        .add_attribute("action", "instantiate")
+                        .add_attribute("token", "existing_token")
+                        .add_attribute("token_address", address)
+                        .add_attribute("staking_contract", staking_contract_address))
+                }
+                StakingInfo::New {
+                    staking_code_id,
+                    unstaking_duration,
+                } => {
+                    let msg = WasmMsg::Instantiate {
+                        code_id: staking_code_id,
+                        funds: vec![],
+                        admin: Some(env.contract.address.to_string()),
+                        label: env.contract.address.to_string(),
+                        msg: to_binary(&stake_cw20::msg::InstantiateMsg {
+                            admin: Some(env.contract.address.to_string()),
```

Comment:
Same thing here. We probably either want to make the DAO the admin or add methods to this contract to let the DAO manage it. Probably simpler to just make the DAO the admin. :)